pkg/odo/cli/service/create.go

```diff
@@ -45,46 +49,96 @@ func NewServiceCreateOptions() *ServiceCreateOptions {
 
 // Complete completes ServiceCreateOptions after they've been created
 func (o *ServiceCreateOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
-	o.serviceType = args[0]
-	// if only one arg is given, then it is considered as service name and service type both
-	o.serviceName = o.serviceType
-	// if two args are given, first is service type and second one is service name
-	if len(args) == 2 {
-		o.serviceName = args[1]
+	if len(args) == 0 || !cmd.HasFlags() {
+		o.interactive = true
```

Comment:
I think interactive should run even if the name and type is there. In that case it should ask for the parameters only. Also a check if all the required parameters are filled up. If a required parameter is missing value, inform and ask the user for the value. WDYT?

Reply:
See the discussion above: it's been agreed that interactive mode only kicks in when there are no flags or parameters. If you believe otherwise (as I do ^_^), please comment to that effect on #76

Reply:
Ok left a comment on the discussion :)